cityofaustin/atd-moped
# Change knack payload sent to Data Tracker to richly populate `link` field

## Associated issues

This PR intends to close [issue 8452](https://github.com/cityofaustin/atd-data-tech/issues/8452).

## Testing
**URL to test:** 

https://deploy-preview-609--atd-moped-main.netlify.app/moped/projects is possible, but a local environment is much easier to use IMHO, to allow for easier manipulation of the underlying database.

**Steps to test:**

In order to test this PR, you'll need a project which does not have a representation in the Data Tracker projects object and the same project to have `null` in its record for `knack_project_id` in the moped database. I've done this most easily running moped locally and removing conflicting rows from the dev Data Tracker and this query to null out the project_knack_id for a given project:

```
update moped_project
set knack_project_id = null
where project_id = <project_id_here>;
```

With the above things, you can test this PR by using the `Synchronize with Data Tracker` function from the project summary page. When complete, check that you find the record which has been created in the Data Tracker app has the appropriate data expected, including and in particular, a correct & rich value in the Moped link field.

This link field should contain both the:
* [ ] URL for the link
* [ ] label to be displayed in the anchor itself

NB: The signal connection field, due to a mismatch of signal IDs in the development v. production data tracker deployments are not expected to work outside of production.

## Deployment Considerations

There are no changes to the production Data Tracker that need to occur to receive the payload generated by these changes.  

---
#### Ship list
